xdsl/utils/immutable_list.py

```diff
@@ -48,25 +49,43 @@ def clear(self) -> None:
             raise Exception("frozen list can not be modified")
         return super().clear()
 
+    @overload
     def __setitem__(self, __index: SupportsIndex, __object: _T) -> None:
+        ...
+
+    @overload
+    def __setitem__(self, __index: slice, __object: Iterable[_T]) -> None:
+        ...
+
+    def __setitem__(
+        self, __index: SupportsIndex | slice, __object: _T | Iterable[_T]
+    ) -> None:
         if self._frozen:
             raise Exception("frozen list can not be modified")
-        return super().__setitem__(__index, __object)
+        if isinstance(__index, slice):
+            o = cast(list[_T], __object)
+            return super().__setitem__(__index, o)
+        else:
+            o = cast(_T, __object)
+            return super().__setitem__(__index, o)
```

Comment:
Another one for your compendium @AntonLydike :)

Reply:
What's the compendium ? ;)